toolchain/check/testdata/interop/cpp/function/pointer.carbon

```diff
@@ -92,7 +141,7 @@ struct S {};
 
 auto foo(S* _Nonnull const) -> void;
 
-// --- fail_todo_import_const_pointer_param.carbon
+// --- import_const_non_nullable_pointer_param_using_const.carbon
```

Comment:
The naming conventions here (for both `.h` and `.carbon`) are kind of disorienting: sometimes the file name says `non_nullable`, sometimes it says `nullable`, and sometimes it doesn't say either (which seems to mean "non-nullable"?). I'd suggest either putting `nullable`/`non_nullable` in every file name, or putting `non_nullable` on all the files that work with non-nullable pointers, and omitting `nullable` on all the others (thinking of them as testing "ordinary" C++ pointers).

Reply:
I made this more consistent, except for the deduced pointer type tests, which are somewhat special.